src/node_http2.cc

```diff
@@ -975,18 +981,19 @@ void Http2Session::OnHeaders(
   // like {name1: value1, name2: value2, name3: [value3, value4]}. We do it
   // this way for performance reasons (it's faster to generate and pass an
   // array than it is to generate and pass the object).
-  do {
+  size_t n = 0;
+  while (count > 0) {
     size_t j = 0;
-    while (!headers->empty() && j < arraysize(argv) / 2) {
-      nghttp2_header item = headers->front();
+    while (count > 0 && j < arraysize(argv) / 2) {
+      nghttp2_header item = headers[n++];
```

Comment:
Can’t this just be `headers[--count]`? I don’t think you really need the `n` if you didn’t need it before